porch/webhooks/main.go

```diff
@@ -0,0 +1,357 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
```

Comment:
What is the benefit of having it separate from porch-server? With them being separate, it seems like there is a race on startup where porch-server might come up before the webhook and therefore we will not have the protection of the webhook for a period of time.

Reply:
Made it a separate port on the porch-server rather than a separate pod.